internal/registry/service.go

```diff
@@ -126,6 +127,13 @@ func (client *Client) GetIteration(ctx context.Context, bucketSlug string, opts
 	}
 
 	if resp.Payload.Iteration != nil {
+		revokeAt := time.Time(resp.Payload.Iteration.RevokeAt)
+		if !revokeAt.IsZero() && revokeAt.Before(time.Now().UTC()) {
+			// If RevokeAt is not a zero date and is before NOW, it means this iteration is revoked and should not be used
+			// to build new images.
+			return nil, fmt.Errorf("the iteration %s is revoked and can not be used on Packer builds",
+				resp.Payload.Iteration.ID)
```

Comment:
I just realized that the error check for revoke_at is in the registry service code and not in the data source code itself. Is there any reason why you went that route as opposed to putting it in the data source?

Reply:
I was just reverting the first implementation and haven't thought too much about but I think it makes sense to put the check in the data source instead! So I updated the PR.